Add nf-core bot to automate changelog, fix linting, update snapshots

jonasscheid opened the pull request:
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

jonasscheid committed:
add nf-core bot automations

jonasscheid committed:
specify mhcquant as target repo

mashehu commented on .github/workflows/changelog.yml:
```diff
@@ -0,0 +1,93 @@
+          PR_NUMBER: ${{ github.event.issue.number || github.event.pull_request.number }}
+          PR_TITLE: ${{ github.event.issue.title || github.event.pull_request.title }}
+        run: |
+          python ${GITHUB_WORKSPACE}/.github/workflows/changelog.py
```
you need to also add this script to make this action work

mashehu commented on .github/workflows/update-snapshots.yml:
these are the wrong snapshots, this tests our TUI.
You are looking for something like https://github.com/nf-core/modules/blob/master/.github/workflows/update-gpu-snapshot.yml

jonasscheid replied:
Thanks for the pointer! I tried to implement it according to this workflow. Is there any way to dry-run it?

jonasscheid replied:
@mashehu Is that what you had in mind?

jonasscheid committed:
Add changelog update script for CI integration

This script updates the CHANGELOG.md based on PR title and number, ensuring entries are added correctly without duplication.

jonasscheid committed:
Implementation attempt to update-snapshots using nf-core bot

FriederikeHanssen commented:
would it make sense to add this to the template?

mashehu commented:
Maybe, but we usually go the: first have it in some pipelines to find edge cases and then add it to the template

jonasscheid commented:
I volunteer 👋🏼 Let's merge this one soon, then I can test

mirpedrol commented on .github/workflows/update-snapshots.yml:
```diff
@@ -0,0 +1,102 @@
+name: Update Textual snapshots from a comment
```
```suggestion
name: Update nf-test snapshots from a comment
```

jonasscheid committed:
Update .github/workflows/update-snapshots.yml

Co-authored-by: Júlia Mir Pedrol <mirp.julia@gmail.com>
